core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java

```diff
@@ -222,34 +294,103 @@ private DriverContext createDriverContext(TaskContext taskContext)
                     .addDriverContext();
         }
 
-        private TaskContext createTaskContext()
+        private TaskContext createTaskContext(Session session)
         {
-            return TestingTaskContext.builder(EXECUTOR, SCHEDULER, TEST_SESSION)
-                    .setMemoryPoolSize(MAX_MEMORY)
-                    .build();
+            return TestingTaskContext.builder(EXECUTOR, SCHEDULER, session).build();
         }
 
-        private PartitionedOutputBuffer createPartitionedBuffer(OutputBuffers buffers, DataSize dataSize)
+        private TestingPartitionedOutputBuffer createPartitionedBuffer(OutputBuffers buffers, DataSize dataSize)
         {
-            return new PartitionedOutputBuffer(
+            return new TestingPartitionedOutputBuffer(
                     "task-instance-id",
                     new StateMachine<>("bufferState", SCHEDULER, OPEN, TERMINAL_BUFFER_STATES),
                     buffers,
                     dataSize,
                     () -> new SimpleLocalMemoryContext(newSimpleAggregatedMemoryContext(), "test"),
-                    SCHEDULER);
+                    SCHEDULER,
+                    blackhole);
+        }
+
+        private static class TestingPartitionedOutputBuffer
+                extends PartitionedOutputBuffer
+        {
+            private final Blackhole blackhole;
+
+            public TestingPartitionedOutputBuffer(
+                    String taskInstanceId,
+                    StateMachine<BufferState> state,
+                    OutputBuffers outputBuffers,
+                    DataSize maxBufferSize,
+                    Supplier<LocalMemoryContext> systemMemoryContextSupplier,
+                    Executor notificationExecutor,
+                    Blackhole blackhole)
+            {
+                super(taskInstanceId, state, outputBuffers, maxBufferSize, systemMemoryContextSupplier, notificationExecutor);
+                this.blackhole = blackhole;
+            }
+
+            // Use a dummy enqueue method to avoid OutOfMemory error
+            @Override
+            public void enqueue(int partitionNumber, List<SerializedPage> pages)
+            {
+                if (blackhole != null) {
```

Comment:
I don't think there is a difference between ignoring page and  passing it to blackhole.
Just keep last serialized page.